Source/Particles/MultiParticleContainer.cpp

```diff
@@ -991,6 +1011,112 @@ MultiParticleContainer::BreitWheelerGenerateTable ()
+        getSpeciesID(m_qed_schwinger_pos_product_name);
+
+   // TODO: test if electrons (positrons) are really electrons (positrons)
+
+}
+
+void
+MultiParticleContainer::doQEDSchwinger ()
+{
+    if (!m_do_qed_schwinger) {return;}
+
+    auto& pc_product_ele =
+            allcontainers[m_ind_qed_schwinger_ele_product];
+    auto& pc_product_pos =
+            allcontainers[m_ind_qed_schwinger_ele_product];
+
+    auto & warpx = WarpX::GetInstance();
+
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(warpx.do_nodal,
+        "ERROR: Schwinger process only implemented for warpx.do_nodal = 1");
+
+    // So far Schwinger process does not consider refined meshes
+    const int level_0 = 0;
+
+    const MultiFab & Ex = warpx.getEfield(level_0,0);
+    const MultiFab & Ey = warpx.getEfield(level_0,1);
+    const MultiFab & Ez = warpx.getEfield(level_0,2);
+    const MultiFab & Bx = warpx.getBfield(level_0,0);
+    const MultiFab & By = warpx.getBfield(level_0,1);
+    const MultiFab & Bz = warpx.getBfield(level_0,2);
+
+        // get cell size multiplied by time step
+    Geometry const & geom = warpx.Geom(level_0);
+    auto domain_box = geom.Domain();
+#if (AMREX_SPACEDIM == 2)
+    auto dVdt = geom.CellSize(0) * geom.CellSize(1) * warpx.getdt(0); // getdt(level_0) ??
+#elif (AMREX_SPACEDIM == 3)
+    auto dVdt = geom.CellSize(0) * geom.CellSize(1) * geom.CellSize(2) * warpx.getdt(0);
+#endif
+
+    // Add defineAllParticleTiles(); ??
+
+    MFItInfo info;
+    if (TilingIfNotGPU()) {
+        info.EnableTiling(); // Put EnableTiling(PhysicalParticleContainer::tile_size); instead ??
+    }
+#ifdef _OPENMP
+    info.SetDynamic(true);
+#pragma omp parallel if (not WarpX::serialize_ics)
+#endif
+
+    for (MFIter mfi(Ex, info); mfi.isValid(); ++mfi )
+    {
+        // Make the box cell centered to avoid creating particles twice on the tile edges
+        const Box& box = enclosedCells(mfi.tilebox());
+
+        // const FArrayBox& fabEx = Ex[mfi];
+
+/*         const auto& arrEx = Ex[mfi].array();
+        const auto& arrEy = Ey[mfi].array();
+        const auto& arrEz = Ez[mfi].array();
+        const auto& arrBx = Bx[mfi].array();
+        const auto& arrBy = By[mfi].array();
+        const auto& arrBz = Bz[mfi].array(); */
+
+        const FArrayBox *array_EMFAB [] = {&Ex[mfi],&Ey[mfi],&Ez[mfi],
+                                           &Bx[mfi],&By[mfi],&Bz[mfi]};
+// To add in filter
+/*         FArrayBox NumPairCreation(box, 1);
+        auto arrNumPairCreation = NumPairCreation.array();
+
+        amrex::ParallelFor(box,  [=] AMREX_GPU_DEVICE (int i, int j, int k){
+                arrNumPairCreation(i,j,k) = getSchwingerProductionNumber( dVdt,
+                    arrEx(i,j,k),arrEy(i,j,k),arrEz(i,j,k),
+                    arrBx(i,j,k),arrBy(i,j,k),arrBz(i,j,k));
+                amrex::Print(arrNumPairCreation(i,j,k));
+                }); */
+
+        auto& dst_ele_tile = pc_product_ele->ParticlesAt(level_0, mfi);
```

Comment:
Are we sure that adding at level 0 is OK for particles?

Reply:
No :) (I'm not sure if the particles will be put at the correct level the next time Redistribute() is called)
Let's assume for now that this module only works without mesh refinement (I've added an assert for that), we can modify this later on.